### ISSUE TITLE
Suggested lint for numeric literals with leading unary negation

### ISSUE THREAD

**Issue by PhilipDaniels** [rust-lang/rust-clippy#4228]
I recently posted a [thread](https://users.rust-lang.org/t/intervening-spaces-are-allowed-in-negative-numeric-literals/29423) on Rust Users when I was tripped up by this piece of code

```rs
if a >- 150 {
```

It's a simple typo (it should be `>=`), but quite difficult to spot. It's not currently warned on by clippy.
A user on the thread did point out that the current behaviour is similar to what happens in other languages, and is not in fact ambiguous with respect to precedence, however, I wondered if people think it worth a lint? It took me an hour or so to find this error.

I can think of two 'problems' with the above that could be converted to a lint:

* The `-` should not be separated from the literal
* `>-` is not a valid operator.

Of the two, the first is probably a better candidate, as I know there is a tendency for some programmmers to eschew whitespace when writing mathematial expressions making things such as `if a>-150 {` or `x*x>90` quite common (perhaps more so in C than in Rust).

The title says unary negation, but I guess `+ 200` ought to be considered as well.

**Comment by flip1995** [comment]
We could add a (pedantic) lint for the first case you described.

But as already pointed out in the forum thread, `rustfmt` automatically formats this to `if a > -150 {` ([Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=991a968501f9f8c4fb7a8bb40dfc8599) example). Since many people run `rustfmt` on save, this lint would not trigger or them and I don't really see another way to detect this.

**Comment by mikerite** [comment]
This is similar to [suspicious_assignment_formatting](https://rust-lang.github.io/rust-clippy/master/index.html#suspicious_assignment_formatting). IMO, it would make sense to call it 'suspicious_comparison_formatting' and put it in the style group.

**Comment by mikerite** [comment]
Or maybe this check can be applied to all binary operations. :thinking:

**Comment by nikofil** [comment]
Hey! I made a PR about this issue to get my feet wet with this project. I hope I understood @mikerite's suggestion about doing this check on all binary operations correctly - this way makes sense to me. All feedback welcome!
